[BUG] FTDI EVE fails to build with AUTO_BED_LEVELING_BILINEAR

BOARD_FYSETC_S6_V2_0
LCD_FYSETC_TFT81050
S6_TFT_PINMAP

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/5747102/Marlin.zip)




Linking .pio/build/FYSETC_S6/firmware.elf
/home/lino/.platformio/packages/toolchain-gccarmnoneeabi@1.90201.191206/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: .pio/build/FYSETC_S6/src/src/lcd/extui/lib/ftdi_eve_touch_ui/marlin_events.cpp.o: in function `ExtUI::onMeshUpdate(signed char, signed char, float)':
/home/lino/Drucker/fysetc_s6/Firmware/Marlin_2.0.8/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/marlin_events.cpp:144: undefined reference to `BedMeshScreen::onMeshUpdate(signed char, signed char, float)'
/home/lino/.platformio/packages/toolchain-gccarmnoneeabi@1.90201.191206/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: .pio/build/FYSETC_S6/src/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/screens.cpp.o:(.rodata._ZN9ScreenRef13functionTableE+0x340): undefined reference to `BedMeshScreen::onEntry()'
/home/lino/.platformio/packages/toolchain-gccarmnoneeabi@1.90201.191206/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: .pio/build/FYSETC_S6/src/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/screens.cpp.o:(.rodata._ZN9ScreenRef13functionTableE+0x350): undefined reference to `BedMeshScreen::onRedraw(draw_mode_t)'
/home/lino/.platformio/packages/toolchain-gccarmnoneeabi@1.90201.191206/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: .pio/build/FYSETC_S6/src/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/screens.cpp.o:(.rodata._ZN9ScreenRef13functionTableE+0x354): undefined reference to `BedMeshScreen::onTouchStart(unsigned char)'
/home/lino/.platformio/packages/toolchain-gccarmnoneeabi@1.90201.191206/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: .pio/build/FYSETC_S6/src/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/screens.cpp.o:(.rodata._ZN9ScreenRef13functionTableE+0x35c): undefined reference to `BedMeshScreen::onTouchEnd(unsigned char)'
collect2: error: ld returned 1 exit status
*** [.pio/build/FYSETC_S6/firmware.elf] Error 1
================================================================================== [FAILED] Took 42.47 seconds ==================================================================================

Environment    Status    Duration
-------------  --------  ------------
FYSETC_S6      FAILED    00:00:42.472
============================================================================= 1 failed, 0 succeeded in 00:00:42.472 =============================================================================
The terminal process "pio 'run', '--environment', 'FYSETC_S6'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.

--- a/Marlin/src/inc/Conditionals_LCD.h
+++ b/Marlin/src/inc/Conditionals_LCD.h
@@ -1045,11 +1045,6 @@
   #define INVERT_E_DIR false
 #endif
 
-// Fallback SPI Speed for SD
-#if ENABLED(SDSUPPORT) && !defined(SD_SPI_SPEED)
-  #define SD_SPI_SPEED SPI_FULL_SPEED
-#endif
-
 /**
  * This setting is also used by M109 when trying to calculate
  * a ballpark safe margin to prevent wait-forever situation.

--- a/Marlin/src/inc/Conditionals_post.h
+++ b/Marlin/src/inc/Conditionals_post.h
@@ -2708,6 +2708,11 @@
   #endif
 #endif
 
+// Fallback SPI Speed for SD
+#if ENABLED(SDSUPPORT) && !defined(SD_SPI_SPEED)
+  #define SD_SPI_SPEED SPI_FULL_SPEED
+#endif
+
 // Defined here to catch the above defines
 #if ENABLED(SDCARD_SORT_ALPHA) && (FOLDER_SORTING || ENABLED(SDSORT_GCODE))
   #define HAS_FOLDER_SORTING 1

--- a/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/bed_mesh_screen.cpp
+++ b/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/bed_mesh_screen.cpp
@@ -21,7 +21,7 @@
 
 #include "../config.h"
 
-#if BOTH(TOUCH_UI_FTDI_EVE, AUTO_BED_LEVELING_UBL)
+#if BOTH(TOUCH_UI_FTDI_EVE, HAS_MESH)
 
 #include "screens.h"
 #include "screen_data.h"

--- a/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/bed_mesh_screen.cpp
+++ b/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/bed_mesh_screen.cpp
@@ -309,17 +309,6 @@ void BedMeshScreen::onMeshUpdate(const int8_t, const int8_t, const float) {
     onRefresh();
 }
 
-bool BedMeshScreen::isMeshComplete(ExtUI::bed_mesh_t data) {
-  for (uint8_t y = 0; y < GRID_MAX_POINTS_Y; y++) {
-    for (uint8_t x = 0; x < GRID_MAX_POINTS_X; x++) {
-      if (isnan(data[x][y])) {
-        return false;
-      }
-    }
-  }
-  return true;
-}
-
 void BedMeshScreen::onMeshUpdate(const int8_t x, const int8_t y, const ExtUI::probe_state_t state) {
   switch (state) {
     case ExtUI::MESH_START:
@@ -327,7 +316,7 @@ void BedMeshScreen::onMeshUpdate(const int8_t x, const int8_t y, const ExtUI::pr
       screen_data.BedMesh.message = screen_data.BedMesh.MSG_NONE;
       break;
     case ExtUI::MESH_FINISH:
-      if (screen_data.BedMesh.count == GRID_MAX_POINTS && isMeshComplete(ExtUI::getMeshArray()))
+      if (screen_data.BedMesh.count == GRID_MAX_POINTS && ExtUI::getMeshValid())
         screen_data.BedMesh.message = screen_data.BedMesh.MSG_MESH_COMPLETE;
       else
         screen_data.BedMesh.message = screen_data.BedMesh.MSG_MESH_INCOMPLETE;

--- a/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/screens.h
+++ b/Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/screens/screens.h
@@ -539,7 +539,6 @@ class StepsScreen : public BaseNumericAdjustmentScreen, public CachedScreen<STEP
         static float getHightlightedValue();
         static void drawHighlightedPointValue();
         static void drawMesh(int16_t x, int16_t y, int16_t w, int16_t h, ExtUI::bed_mesh_t data, uint8_t opts, float autoscale_max = 0.1);
-        static bool isMeshComplete(ExtUI::bed_mesh_t data);
 
       public:
         static void onMeshUpdate(const int8_t x, const int8_t y, const float val);

